packages/opal-client/opal_client/policy_store/opa_client.py

```diff
@@ -11,18 +11,31 @@
     BasePolicyStoreClient,
     JsonableValue,
 )
+from opal_client.policy_store.options import WaitStrategy
 from opal_client.utils import proxy_response
 from opal_common.git.bundle_utils import BundleUtils
 from opal_common.opa.parsing import get_rego_package
 from opal_common.schemas.policy import DataModule, PolicyBundle
 from opal_common.schemas.store import JSONPatchAction, StoreTransaction, TransactionType
 from pydantic import BaseModel
-from tenacity import retry, stop_after_attempt, wait_fixed
+from tenacity import retry, stop_after_attempt, wait_exponential, wait_fixed
 
 JSONPatchDocument = List[JSONPatchAction]
 
-# 2 retries with 2 seconds apart
-RETRY_CONFIG = dict(wait=wait_fixed(2), stop=stop_after_attempt(2))
+
+def fetchRetryConfig():
```

Comment:
Since this logic is exclusive to parsing the configuration, I think it would be more correct as part of [policy_store/options.py](https://github.com/permitio/opal/pull/282/files#diff-df6793d1f838d0540eb84920a10ef2444623abf4b1a24a3e520fd7852e8d21b5)

Perhaps, as helper static method of the `PolicyStoreConnRetryOptions` class itself.
It can of course receive the config value itself as a variable

Reply:
I may have misunderstood you. Is 96d5f799930e92c67e2b248668a0ea374e42bdd5 what you had in mind? (creating the tenacity configuration in the `options.py` file)